packages/next/src/server-build.ts

```diff
@@ -1395,6 +1440,7 @@ export async function serverBuild({
                 'x-nextjs-matched-path': '/$1',
               },
               continue: true,
+              override: true,
```

Comment:
Should the override be gated behind `NON_NESTED_MIDDLEWARE_VERSION`?

Reply:
This is already gated behind `isNextDataServerResolving`